### PR TITLE
Applying changes based on opennmt-tf #432

### PR DESCRIPTION
**Pull request by helson73**
Applying changes to Transformer based on opennmt-tf #432

### CODE REVIEW

**Comment by helson73** [comment]
Following changes were applied to Transformer.py and MultiHeadedAttn.py.
1. Move norm layer to the block input.
2. Apply dropout after ReLU in point-wise feed forward block.
3. Add norm layer to encoder and decoder output.

**Comment by helson73** [comment]
Trained on small dataset (WAT 2016 Japanese-to-English, randomly selected 100K sentence pairs) for 20 epochs,
got following result.
**Before:**
Train perplexity: 32.4537
Train accuracy: 38.7784
Validation perplexity: 56.2081
Validation accuracy: 33.0021

**After:**
Train perplexity: 29.8781
Train accuracy: 40.5064
Validation perplexity: 52.0337
Validation accuracy: 34.0583

**Comment by helson73** [comment]
Referred source:

- https://github.com/OpenNMT/OpenNMT-tf/blob/master/opennmt/utils/transformer.py
- https://github.com/OpenNMT/OpenNMT-tf/blob/master/opennmt/encoders/self_attention_encoder.py
- https://github.com/OpenNMT/OpenNMT-tf/blob/master/opennmt/decoders/self_attention_decoder.py

**Comment by srush** [comment]
Awesome, thanks for testing this. I guess the two next steps are 

1) batch size by tokens
2) label smoothing.
